torch/_higher_order_ops/associative_scan.py

```diff
@@ -102,9 +113,6 @@ def add(x: torch.Tensor, y: torch.Tensor):
     return pytree.tree_unflatten(result_flat, spec)
 
 
-associative_scan_op = HigherOrderOperator("associative_scan")
```

Comment:
okay we should probably ban creating HigherOrderOp without subclassing it

Reply:
Will do it in a separate diff.